analysis.py

```diff
@@ -36,7 +37,56 @@ def visualize_clusters(directory, bins=None, plot_kwargs={}, save_kwargs={'dpi':
     fig.savefig(os.path.join(directory, 'electron-energy-vs-ion_energy-difference.png'), **save_kwargs)
     plt.close(fig)
 
-def optimal_angular_distribution_hyperparameters(train_data, save_dir, val_data, labels, train_indices, val_indices, L_max, bin_range):
+def optimal_k_means_hyperparameters(phi, data, train_data, val_data, train_indices, val_indices, cluster_range, L_max, num_bins, save_dir='.'):
```

Comment:
We should start commenting these functions so that we know what each does; even though this is not source files, it would still be helpful.

Reply:
yeah ok